File: idleAFK.js
const Discord = require('discord.js'); // you'll have to convert to discord.js-selfbot-v13, I am inexperienced in that area.
const client = new Discord.Client();

const idleTimeout = 30 * 60 * 1000;
let lastMessageTimestamp = Date.now();

client.on('ready', () => {
    console.log(`Logged in as ${client.user.tag}`);
});

client.on('message', (message) => {
    if (message.author.bot) return;

    lastMessageTimestamp = Date.now();

    client.user.setPresence({ status: 'online' });
});

function checkIdleStatus() {
    const currentTime = Date.now();
    const elapsedTime = currentTime - lastMessageTimestamp;

    if (elapsedTime >= idleTimeout) {
        client.user.setPresence({ status: 'idle' });
    }
}

setInterval(checkIdleStatus, 60 * 1000);

client.login('TOKEN');

File: idleAFK.py
import discord
from discord.ext import tasks
from datetime import datetime, timedelta

client = discord.Client()

last_message_time = None

@client.event
async def on_ready():
    print(f'Logged in as {client.user.name}')
    check_idle.start()

@client.event
async def on_message(message):
    global last_message_time
    if message.author == client.user:
        last_message_time = datetime.now()
        await client.change_presence(status=discord.Status.online)

@tasks.loop(minutes=1)
async def check_idle():
    global last_message_time
    if last_message_time is not None:
        idle_time = timedelta(minutes=30)
        if datetime.now() - last_message_time >= idle_time:
            await client.change_presence(status=discord.Status.idle)


client.run('TOKEN')